python/ray/data/iterator.py

```diff
@@ -256,28 +255,42 @@ def iter_torch_batches(
         batch_size: Optional[int] = 256,
         dtypes: Optional[Union["torch.dtype", Dict[str, "torch.dtype"]]] = None,
         device: Optional[str] = None,
-        collate_fn: Optional[
-            Callable[[Union[np.ndarray, Dict[str, np.ndarray]]], Any]
-        ] = None,
+        collate_fn: Optional[Callable[[Dict[str, np.ndarray]], Any]] = None,
         drop_last: bool = False,
         local_shuffle_buffer_size: Optional[int] = None,
         local_shuffle_seed: Optional[int] = None,
         # Deprecated.
         prefetch_blocks: int = 0,
-    ) -> Iterator["TorchTensorBatchType"]:
+    ) -> Iterator[Dict[str, "torch.Tensor"]]:
         """Return a batched iterator of Torch Tensors over the dataset.
 
-        This iterator will yield single-tensor batches if the underlying dataset
-        consists of a single column; otherwise, it will yield a dictionary of
-        column-tensors. If looking for more flexibility in the tensor conversion (e.g.
-        casting dtypes) or the batch format, try using `.iter_batches` directly.
+        This iterator will will yield a dictionary of column-tensors. If looking for
+        more flexibility in the tensor conversion (e.g. casting dtypes) or the batch
+        format, try using `.iter_batches` directly.
 
         Examples:
             >>> import ray
-            >>> for row in ray.data.range(
-            ...     1000000
-            ... ).iterator().iter_rows(): # doctest: +SKIP
-            ...     print(row) # doctest: +SKIP
+            >>> for batch in ray.data.range(
+            ...     12,
+            ... ).iterator().iter_torch_batches(batch_size=4):
+            ...     print(batch.shape)
+            {'id': tensor([ 0, 10,  1,  2])}
+            {'id': tensor([3, 4, 5, 6])}
+            {'id': tensor([ 8,  9, 11,  7])}
+
+            Use the ``collate_fn`` to customize how the tensor batch is created.
+            >>> from typing import Any, Dict
+            >>> import torch
+            >>> import numpy as np
+            >>> import ray
+            >>> def collate_fn(batch: Dict[str, np.ndarray]) -> Any:
+            ...     return torch.stack(
+            ...     [torch.as_tensor(array) for array in batch.values()],
+            ...     axis=1)
+            >>> iterator = ray.data.from_items([{"col_1": 1, "col_2": 2}]).iterator()
```

Comment:
Can we use more than 1 items here? So that we can apply `collate_fn` on a more realistic batch.

Reply:
updated